From e66769ad5194cb4533d1087416a2e804ac384285 Mon Sep 17 00:00:00 2001
From: Kenny Root <kroot@google.com>
Date: Mon, 4 Feb 2013 15:49:11 -0800
Subject: [PATCH 279/283] AndroidKeyStore: return error code on error

Instead of blindly multiplying return value by 1000 to convert to
milliseconds, check to see if it's an error condition first.

Change-Id: I8eab1e7a86d78c13458fcbbc79d590e452fc9791
---
 keystore/java/android/security/KeyStore.java | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/keystore/java/android/security/KeyStore.java b/keystore/java/android/security/KeyStore.java
index 444dc1d..ceaff37 100644
--- a/keystore/java/android/security/KeyStore.java
+++ b/keystore/java/android/security/KeyStore.java
@@ -243,7 +243,12 @@ public class KeyStore {
      */
     public long getmtime(String key) {
         try {
-            return mBinder.getmtime(key) * 1000L;
+            final long millis = mBinder.getmtime(key);
+            if (millis == -1L) {
+                return -1L;
+            }
+
+            return millis * 1000L;
         } catch (RemoteException e) {
             Log.w(TAG, "Cannot connect to keystore", e);
             return -1L;
-- 
1.8.1.1

